my-app/src/main/java/com/JavaWebServer/app/Responder.java

```diff
@@ -35,19 +31,27 @@ public Responder(Map routeDirectory, Map routes) {
       return BAD_REQUEST.getBytes();
     }
   }
-  
+
   private byte [] getMessage(Request request){
-    String message = request.getRequest(); 
+    String message = request.getRequest();
     RestMethod currentRoute = routes.get(message);
-    return currentRoute.handleRequest();
+    return currentRoute.handleRequest(request);
   }
 
   private boolean checkRoute(Request request) {
-    return routeDirectory.containsKey(request.getRoute());
+   if (request.isParams()) {
```

Comment:
It doesn't seem very extensible to have to check if the Request has routes just to get the path. Can you think of a better way to abstract out the different ways to get a requests route?